Clone estimator for each parameter value in validation_curve

Sundrique opened the pull request:
In validation_curve function an estimator is not cloned like in learning_curve or GridSearchCV.fit for each value of a parameter. Thus it comes pre-trained to each iteration but very first.

ndingwall committed:
Adds support for step-wise interpolation to auc and average_precision_score

amueller commented:
thanks for the fix. Do you think you can create a non-regression test?

Sundrique commented:
I think, I can if it can wait a couple of days.

amueller commented:
sure, no worries :)

Sundrique committed:
Add test to verify validation_curve clones estimator

jnothman commented:
I find your mock estimator a bit of an obtuse way to test this. How about a class inheriting from something already in scikit-learn (even a dummy estimator) that simply raises an error if fit is called on it multiple times?
```
    def fit(self, X, y):
        assert_false(hasattr(self, 'fit_called_'))
        self.fit_called_ = True
        super(type(self), self).fit(X, y)
```

ogrisel commented:
Please also make sure that the lines you change in this PR follow the PEP8 conventions (travis is read because of this):

https://travis-ci.org/scikit-learn/scikit-learn/jobs/160421906#L457

Sundrique commented:
Thanks @jnothman, @ogrisel. Will fix.

Sundrique committed:
Simplify test and mock estimator

Sundrique committed:
Reformat to reduce lines length

jnothman commented:
Actually, this change needs to be in `model_selection/_validation.py`, although I think it's a good idea to backport it to the deprecated module you've currently edited. Otherwise, LGTM. (And I confirm the test fails at master.)

Commit:
Update docs with new line_profiler signature and extension method

Update memory_profiler docs as well

Append new extensions instead of setting them directly

mathurinm committed:
Correct figure number + matplotlib 2 (#8483)

jnothman committed:
DOC example of extracting true positive, false negative, etc. (#8469)

almer-t committed:
DOC correct typo in kneighbors parameter documentation. (#8495)

jakevdp committed:
[MRG+1] BUG: fix svd_solver validation in PCA.fit (#8496)

* BUG: fix svd_solver validation in PCA.fit

* TST: add test of pca svd_solver

zorroblue committed:
Added check_X_y to lasso_stability_path() (#7534)

aabadie committed:
codecov: disable comments (#8502)

rishikksh20 committed:
[MRG] Modify Svm examples for matplotlibv2 comp (#8456)

* Modify plot_custom_kernel for matplotlib v2 comp

Add `edgecolors` attribute in scatter plot for better visualization
in matplotlib version 2

Issue: #8364

* Modify plot_oneclass.py for matplotlib v2 comp

Add `edgecolors` attribute to scatter plot for better
visualization in matplotlib version 2

Issue: #8364

* Modify plot_rbf_parameters for matplotlib v2

Add `edgecolors` attribute to scatter plot for
better visualization.

Issue: #8364

* Modify plot_separating_hyperplane_unbalanced for matplotlib v2

Add `edgecolors` attribute to scatter plot for better visualization.

Issue: #8364

* Modify plo_svm_kernels for matplotlib v2

Add `edgecolors` attribute to scatter plot for better
visualization.

Issue: #8364

* Modify plot_svm_margin for matplotlib v2 comp

Add `edgecolors` attribute to scatter plot for better
visualization.

Issue: #8364

* Modify plot_svm_nonlinear for matplotlib v2

Add `edgecolors` attribute to scatter plot for matplotlib
version 2 compatibility

Issue: #8364

* Modify file for remove flake8 error

Remove extra white space.

Issue: #8364

amueller committed:
turn comments off in codecov

EFerriss committed:
add html-noplot and changed help message to make.bat (#8524)

* add html-noplot and help message to make.bat

* changed spaces to tab in make.bat help

* changed all spaces to tabs in make.bat update

neerajgangwar committed:
[MRG+1] Repeated K-Fold and Repeated Stratified K-Fold (#8120)

* Add _RepeatedSplits and RepeatedKFold class

* Add RepeatedStratifiedKFold and doc for repeated cvs

* Change default value of n_repeats

* Change input parameters of repeated cv constructor to n_splits, n_repeats, random_state

* Generate random states in split function rather than store it beforehand

* Doc changes, inheriting RepeatedKFold, RepeatedStratifiedKFold from _RepeatedSplits and other review changes

* Remove blank line, put testcases for deterministic split in loop and add StopIteration check in testcase

* Using rng directly as random_state param to create cv instance and added a check for cvargs

* Fix pep8 warnings

* Changing default values for n_splits and n_repeats and add entry in changelog

* Adding name to the feature

* Missing space

Ellen-Co2 committed:
modify disadvantage (#8521)

[MRG+2] modify disadvantage

Namiya committed:
fix deprecated comparison to string in GP (#8518)

Commit:
[MRG+2] referred reliability diagrams and added citations (#8527)

nikitasingh981 committed:
[MRG] Separated regression metrics from other metrics in test_sample_weight_invariance in metrics/tests/test_common.py  (#8537)

* Separated tests for regression features in test_sample_weight_invariance

* Fixed pep8

* Removed unecessary check for regression

* Updated regression metrics

* Joel's suggestions

Commit:
[MRG] removed download_url from setup.py (#8513)

Commit:
[MRG+1] Fixes #7578 added check_decision_proba_consistency in estimator_checks (#8253)

brownsarahm committed:
[MRG+2] addresses #8509 improvements to f_regression documentation (#8548)

* clarify role of the function and streamline introduction

* added feature selection methods to see also

* completed see also

* fixed pep related formatting for flake8checks.

* fixed extra whitespace flake8 problems, remaining failure is a copied see all line from another function, the line is over by a period, does not make sense to newline that.

* one more whitespace

* FIX small pep8 error.

lesteve committed:
[MRG] Update joblib to 0.11 (#8492)

Use pip rather than easy_install in copy_joblib.sh. Also need to remove joblib/testing.py to avoid pytest dependency.

tfolkman committed:
[MRG] DOC More detailed pull request and fork instructions (#8530) (#8538)

matthew-brett committed:
[MRG+2] use manylinux dev wheels for numpy / scipy (#8536)

* MAINT: use manylinux dev wheels for numpy / scipy

Use daily manylinux wheels for numpy and scipy, instead of
soon-to-be-discontinued per-commit Precise wheels.

* BF: add back ATLAS install for ubuntu build entry

scikit-learn can link against BLAS libraries still at the same location
as they were duing numpy build.

chenhe95 committed:
[MRG + 1] Fix gradient boosting overflow and various other float comparison on == (#7970)

* reintroduced isclose() and flake8 fixes to fixes.py

* changed == 0.0 to isclose(...)

* example changes

* changed back to abs() < epsilon

* flake8 convention on file

* reverted flake8 fixes

* reverted flake8 fixes (2)

* np.finfo(np.float32).tiny instead of hard coded epsilon 1e-150

* reverted to 1e-150

* whats new modified

lmcinnes committed:
[MRG+1] DOCATHON : Provide more intuition on perplexity in the documentation,… (#8551)

icyblade committed:
DOC fix typo (#8560)

lesteve committed:
Remove unneeded ** in CONTRIBUTING.md

lesteve committed:
DOC better link in CONTRIBUTING.md (#8561)

glemaitre committed:
FIX/TST fix bug variable name for pure test (#8562)

katelie committed:
[MRG+1] add edgecolor to plot_pca_iris.py (#8514)

for better rendering with matplotlib 2

rishikksh20 committed:
[MRG] remove "matplotlib <1" compatibility code from manifold  examples  (#8552)

ja9harper committed:
[MRG + 1] Correct typo in cross decomposition example (Fixes #8307) (#8578)

* changed plsca to cca

* corrected variable plsca in line 56-57

Trion129 committed:
[MRG+1] Added v2 Custom Search API and fixed css placement for search box (#8542)

* Added v2 Custom Search API and fixed css placement for search box

* Tweaked CSS and hid the branding section

* Button made cute and override some css to make it look good for old api and new api alike

* Better button placement

* Colored the search results and pagination section

leereeves committed:
[MRG+1] Improved docstring for permutation_test_score (#8379 and #8564) (#8569)

rishikksh20 committed:
Correct formulas in Doc of Gaussian Process Kernels (#8571)

Correct the Rational quadratic kernel and ExpSineSquare
kernel formulas.

Issue: #8553

rishikksh20 committed:
[MRG] Modify Classification module example matplotlibv2 (#8516)

ogrisel committed:
FIX s/algorithm/solver/ in bench_mnist.py

glemaitre committed:
[MRG + 2] EHN additional test for trees regarding fitting behaviour with constant features (#8580)

* TST add test checking the behaviour of constant/no-constant features

* FIX/TST factorize test

* TST Add additional constant features

* FIX/TST remove ExtraTree from test

Commit:
[MRG+1] FIX Correct depth formula in iforest  (#8576)

* Fixed depth formula in iforest

* Added non-regression test for issue #8549

* reverted some whitespace changes

* Made changes to what's new and whitespace changes

* Update whats_new.rst

* Update whats_new.rst

* fixed faulty whitespace

* faulty whitespace fix and change to whats new

* added constants to iforest average_path_length and the according non regression test

* COSMIT

* Update whats_new.rst

* Corrected IsolationForest average path formula and added integer array equiv test

* changed line to under 80 char

* Update whats_new.rst

* Update whats_new.rst

* reran tests

* redefine np.euler_gamma

* added import statement for euler_gammma in iforest and test_iforest

* changed np.euler_gamma to euler_gamma

* fix small formatting issue

* fix small formatting issue

* modified average_path_length tests

* formatting fix + removed redundant tests

* fix import error

* retry remote server error

* retry remote server error

* retry remote server error

* re-added some iforest tests

* re-added some iforest tests

gedeck committed:
[MRG + 1] Return correct ridge parameter alpha_ and lambda_ for Bayesian ridge regression (#8567)

* Return correct ridge parameter alpha_ and lambda_ for regression

* Add test for coefficients and fix style

* Move sklearn.utils.testing to a more reasonable position.

* Make flake8 happy

* Code cleanup and entry in whats_new.rst

scooler committed:
Fix typo in documentation (#8600)

x_i should be in R^m (as n is the number of features)

hbertrand committed:
[MRG+1] GaussianProcessRegressor: faster prediction of std (#8591)

Sentient07 committed:
Improve warning message in ElasticNet (#8565)

m1so committed:
Fix typo in grid_search.rst (#8609)

sid-kap committed:
DOC neighbors.rst: use \le and \ge (#8610)

Use \le and \ge instead of <= and >=.

sid-kap committed:
Change `>>` to `\gg` (#8612)

Use the LaTeX symbol for "much greater than"

guiniol committed:
[MRG+1] Fix float size in as_float_array (#8598)

* Fix float size in as_float_array

* Add tests for small ints in as_float_array

* Add test for object dtype

with minor tweaks

gte620v committed:
[MRG+1] [DOC] Adding GMM to plot_cluster_comparison.py (#6305)

* Adding GMM to plot_cluster_comparison.py and changing number of components in all algos to 3.

* adding two datasets to clustering comparision example

* Adding GMM to plot_cluster_comparison.py and changing number of components in all algos to 3.

* adding two datasets to clustering comparision example

* GMM example using GaussianMixture

* fixing lint errors; changing order of datasets in the columns so that no_structure is at the end.

* adding warning supression.

* fixing warning supression.

* hand-tuned cluster parameters

* moved list of algo names; cleaning up color cycling

* fixing islice stop to be an int

* change default to params, make plot color-blind compatible, fix spelling error

* new color palette that is more color-blind friendly

hiqua committed:
Fix typo in BaseMixture warning (#8616)

lmcinnes committed:
DOC: hdbscan is now part of scikit-learn-contrib; update link (#8629)

jnothman committed:
[MRG+2] Fixes kl_divergence_ update in TSNE (#8634)

tboucher committed:
DOC: Fixed year for Ng et al. citation within spectral embedding. (#8643)

qinhanmin2014 committed:
[MRG] DOC improve example plot_forest_iris.py (#8649)

davidheryanto committed:
[MRG] Fix plot_unveil_tree_structure.py: incorrect indexing for X_test (#8653)

Commit:
Add SAGA solver for LogisticRegression and Ridge (#8446)

glemaitre committed:
[MRG] DOC add pre-processing section in related projects (#8648)

lemurey committed:
[MRG] DOC: improve docstring form 'normalize' (#8658)

* DOC: improve docstring form 'normalize'

* DOC: improve docstring for 'normalize'

* fix double backticks in coordinate descent

* add missing path reference

* fix line lengths

* align language in docstring across files in linear_model

weber-s committed:
[DOC] Modify default normalize from False to True for some linear model (issue #1650) (#8640)

martin-hahn committed:
[MRG+1] Calculate confidence intervall only if we have enough samples (#8621)

* calculate confidence intervall only if we have enough samples

* pep8

* removed parentheses

herilalaina committed:
[MRG+1] Change named_steps to Bunch object (#8586)

* Change named_steps to Bunch object

* Update named_steps attribute documentation

* Add test for named steps bunch object

* Delete whitespace in test_pipeline

* Update test_pipeline.py

* Add comment for named_steps usage

* Move dataset/Bunch to utils

* Fix to PEP8 format

* Add __getattribute method to Bunch class, Fix pep8 bug

* Remove __getattribute__, update test_pipeline

* Update test with conflict and non-conflict named_steps

* Add reference to class Pipeline

glemaitre committed:
[MRG+1] DOC remove repetition in the Pipeline/memory doc (#8669)

* DOC remove repetition in the Pipeline/memory doc

* DOC fix comments loic

zxcvbnius committed:
[MRG+1] Fix test of SingleInheritanceEstimator to not raise DeprecationWarning (#8526)

* Fix test of SingleInheritanceEstimator to not raise DeprecationWarning

* [MRG+1] Fix test of SingleInheritanceEstimator to not raise DeprecationWarning (#8526)

Fix: test of SingleInheritanceEstimator to not raise DeprecationWarning (#8526)

- Ignore a DeprecationWarning about unpickling an estimator from a different version

* [MRG+2] Fix test of SingleInheritanceEstimator to not raise DeprecationWarning (#8526)

Fix: test of SingleInheritanceEstimator to not raise DeprecationWarning (#8526)

  - Test of SingleInheritanceEstimator would raise UserWarning, not DeprecationWarning
  - Ignore a UserWarning about unpickling an estimator from a different version

Commit:
Fix typo: "continious" (#8685)

Change "continious" to "continuous" in the "See also" section of the docstring of `SelectKBest`

jnothman committed:
DOC add blank line for RST rendering

chkoar committed:
DOC: Correct scikit-learn name in comments (#8681)

mikebenfield committed:
[MRG+1] Fix excessive memory usage in random forest prediction (#8672)

[MRG+2] Fix excessive memory usage in random forest prediction

raghavrv committed:
[MRG+1] ENH/FIX Introduce min_impurity_decrease param for early stopping based on impurity; Deprecate min_impurity_split (#8449)

[MRG+2] ENH/FIX Introduce min_impurity_decrease param for early stopping based on impurity; Deprecate min_impurity_split

glemaitre committed:
DOC removed deprcated residues_ (#8691)

glemaitre committed:
[MRG+1] DOC improve description and consistency of random_state (#8689)

* DOC improve description of random_state in train_test_split

* DOC Make random_state consistent through documentation

* FIX reverse doc mistake

* FIX address comment of Tom

* DOC address comments

* DOC remove empty line

* DOC remove unecessary white spaces

VathsalaAchar committed:
[MRG+1] OneClassSVM predict now returns int (#8711)

* Added predict method to class OneClassSVM

- This method overrides the default behaviour to only return integer class values.

* Test for the change in OneClassSVM predict method

- small addition to test that the predicted result is integer

* Changed test to check for intp datatype

* Updated whats new

glemaitre committed:
FIX/PEP8 (#8718)

csizsek committed:
[MRG] Notes about default params for trees (#8721)

* Added note to trees and random forests to warn users about the default parameters for controlling tree sizes.

* Rephrased tree size warning.

* Rephrased tree warnings again.

* Added note to ExtraTree* algorithms.

yl565 committed:
[MRG+1] Replace or remove VotingClassifier estimators by set_params (#7674)

* PR to 7288
Use _BaseComposition as base

*  Fix flakes problem

* Change ``pipeline``, add more tests and other changes
1. Use ``_BaseComposition`` in class ``Pipeline`` and ``FeatureUnion``
2. Add tests of soft voting ``transform`` when one estimator is set to None
3. Add estimator name validation in ``_BaseComposition`` and tests
4. Other requested changes.

* Remove the unused import warn

* Add more test and documentation

* resolve conflict with master

* Add testing cases and modify documentation

* Add to whats_new.rst

*  Fix too many blank lines

nalepae committed:
[MRG] grid_search_digits.py: Remove impact-less (and confusing) 'C=1' parameter. (#8731)

This parameter does not modify at all the behavior of GridSearchCV,
and is quite confusing for the user.
Useful values of 'C' are already set in 'tuned_parameters'.

rebeccabilbro committed:
[MRG + 1] Documentation: Adds yellowbrick to related projects (#8733)

* DOCS adds yellowbrick to Related Projects

* DOCS adds Yellowbrick library to Related Projects

Commit:
Fix typo in GridSearchCV documentation (#8739)

TomDLT committed:
add python parameter to issue template for better code rendering

chayant committed:
Fixes tildes in rst files (#8751)

glemaitre committed:
[MRG] DOC change rebase to merge policy (#8753)

* DOC change rebase to merge policy

* FIX remove no-ff flag

puneetmathurDS committed:
[MRG+1] Spelling mistake in Feature Extraction documentation (#8763)

* task: Fixed the minor change in code from svr to svm merging back to main branch http://scikit-learn.org/dev/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV done on: Mar 13 2017 14:30 IST

* issue#8761: Fixed the code with spelling mistake Spelling mistake in
Feature Extraction documentation #8761

hrjn committed:
[MRG+1] Fix multi-label issues in IsolationForest benchmark (#8638)

* Fixed a legacy multi-label issue and added minor refactoring and changes (mostly esthethic)

Minor corrections after code review.

Minor corrections after 2nd code review.

Minor modif

* rerun CI

lesteve committed:
[MRG] Fix sklearn.metrics.classification._check_targets where y_true and y_pred are both binary but the union is multiclass (#8377)

* Fix _check_targets where y_true and y_pred are both binary

but the union of them is multiclass.

* Add entry in changelog

andreh7 committed:
DOC docstring of rand_int value range (#8777)

mppaskov committed:
[MRG+1] Misleading gamma description (#8699) (#8716)

[DOC] update description of gamma for kernel methods

qinhanmin2014 committed:
[MRG] Improve the error message of export_graphviz if a not-fitted decision tree is provided (#8776)

qinhanmin2014 committed:
MAINT Remove unused condition (#8778)

jorisvandenbossche committed:
DOC: fix rst reference to MLPClassifier (#8790)

glemaitre committed:
FIX set min_impurity_split to None in gradient boosting estimator (#8789)

minghui-liu committed:
Removed DataConversionWarning in Normalize (#8712)

AishwaryaRK committed:
Improve error message for MeanShift.estimate_bandwidth when X is sparse #8627 (#8771)

* raise error on sparse matrix in estimate_bandwidth of mean_shift

* Remove added newline

aikinogard committed:
[MRG+1] check length of feature_names in export_graphviz (#8477) (#8512)

* #8477

check length of feature_names in export_graphviz

- raise ValueError if len(feature_names) > tree.n_features
- add unit test for len(feature_names) > tree.n_features
- change the comment of existing unit test for len(feature_names) <
tree.n_features

* fix error and warning

- include length of feature_names and number of features in tree in the
error and warning message.
- raise error for too few feature_names
- for too much feature_names, will use the first n_features. raise an
warning for users
- use assert_raise_message and assert_warns_message in test to check
message.

* move the error and warning from node_to_str to export_graphviz so it will fail early for wrong length of feature_names

* raise error if length of feature_names does not match number of features in the decision tree

* fix pep8

* remove unused assert_warns_message import in test_export.py

* add bug fix in doc/whats_new.rst

* fix the english in doc/whats_new.rst

* fix the format and english in sklearn/tree/export.py

* fix contributor format in doc/whats_new.rst

* fix english, use bracket and avoid \ in error message

* fix pep8

* fix pep8

No-Doy committed:
fixed spelling on comments in file (#8801)

AishwaryaRK committed:
[MRG] Fixes #8736 add get_n_splits for RepeatedKFold and RepeatedStratifiedKFold (#8802)

glemaitre committed:
[MRG+1] EHN add decimals parameter for export_graphviz (#8698)

* EHN add decimals parameter for export_graphviz

* FIX address comments

* TST add test for classification

* TST/FIX address comments

* FIX comments raghav

aivision2020 committed:
[MRG+1] FIX bug where ransac is running too many iterations (#8271)

garyForeman committed:
[MRG] Change deprecation warning for doc_topic_distr from future to present… (#8146)

lmcinnes committed:
[DOC] Fix latex in trustworthiness docstring (#8832)

Curly parens were misplaced resulting in an empty sum when rendered.

chyikwei committed:
DOC update LatentDirichletAllocation `components_` description (#8805)

glemaitre committed:
[MRG+1] ENH add memory to make_pipeline (#8831)

[MRG+2] ENH add memory to make_pipeline

nikitasingh981 committed:
[MRG] Updated title and versions supported in contributing documentation (#8824)

* Changed title and updated versions supported

* Added back a line

* Fixed definition of R^2

* Fixed 80 character limit!

jaroslaw-weber committed:
[MRG+1] added DESCR attribute in diabetes data (#8840)

*  added DESCR attribute in diabetes data

*  added DESCR attribute in diabetes data (fix flake8)

*  added DESCR attribute in diabetes data (added test)

Commit:
[MRG] Extended explanation of using class_weight in RandomForestClassifier (Issue #6646) (#8838)

* Extended explanation of using class_weight in RandomForestClassifier

* Extended explanation of using class_weight in DecisionTreeClassifier,ExtraTreesClassifier and compute_sample_weight()

* Rephrased description.

* Rephrased description (remove "indicator")

manrajgrover committed:
[MRG] Randomized Search Example: Fix min_samples_split values for RandomizedSearchCV and GridSearchCV (#8847)

* Randomized Search Example: Fix min_samples_split values for RandomizedSearchCV and GridSearchCV

* Randomized Search Example: Change min_samples_split from 1.0 -> 2

lesteve committed:
DOC fix rst link

lesteve committed:
DOC fix more rst links

naoyak committed:
[MRG] Always download latest Miniconda on Travis (#8857)

lesteve committed:
Remove miniconda installer caching (#8859)

Download takes less than a second both on Travis and Circle.

Also use latest miniconda installer on Circle.

naoyak committed:
[MRG] Update URL for species distribution dataset (#8864)

lesteve committed:
[MRG] Run more examples as part of the doc generation (#8851)

Also fix errors and flake8 violations of existing examples

edrogers committed:
[MRG] Documenting char_wb padding functionality (Issue #8694) (#8803)

* Documenting char_wb padding functionality (Issue #8694)

* Small fix: change of wording.

* 's/passed with space/padded with space/g'

glemaitre committed:
EXA improve examples (#8866)

Commit:
[MRG+1] Fix inverse_transform in deprecated GridSearchCV (#8860)

tkamishima committed:
[MRG+1] enable to use get_n_splits of LeaveOneGroupOut and LeavePGroupsOut with dummy parameters (#8794)

* remove needless argument checking

* add parameter checking as in LeavePGroupsOut

* add examples with dummy inputs

* add unittest for a get_n_splits method in LeaveOneGroupOut and LeavePGroupsOut classes

* X and y can be ommited in a get_n_splits function.

* fix error messages

* update examples

* fix test for an error message

* Revert "fix test for an error message"

This reverts commit 68b984207c704de1a3411d9ded68a11eba1e56f3.

* fix test for an error message

* fix error messages

* remove tailing white spaces

* add periods to messages

* test for ValueError’s of get_n_splits methods of LeaveOneOut / LeavePOut classes

* fix documents:
* parameter name: group -> groups
* modfy white space

rubik committed:
DOC: fix typos in linear_model.rst (#8868)

* DOC fix typos in linear_model.rst

* Another typo

chkoar committed:
DOC Fix user guide link in MultiTaskElasticNetCV docstring (#8880)

rubik committed:
DOC: add missing default value in LinearRegression docstring (#8873)

* DOC fix typos in linear_model.rst

* Another typo

* DOC: add missing default value in LinearRegression docstring and fix typo

amueller committed:
minor fixes to whats_new.rst for 0.19-dev (#8886)

glemaitre committed:
[MRG+1] EHN/DOC Make error msg about Memory more explicit (#8865)

Commit:
Add information what +1 and -1 means (#8891)

Commit:
relax skewness assumption (#7573)

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

last corrections

whats_new

merge

complying whats_new

removed unnecessary _assert_X

increased coverage

relax skewness assumption

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

relax skewedness assumption

last corrections

whats_new

merge

complying whats_new

removed unnecessary _assert_X

increased coverage

remove cythonize.dat and merge

simplify tests

simplify tests

sergulaydore committed:
[MRG+1] removed precomputed support in nearestcentroid (#8515)

* removed test_precomputed

* Revert "removed test_precomputed"

This reverts commit 71ace3a39ce7d5ad72ecaa1ae2b762d6d7cb6c3e.

* raise an error if metric is precomputed

* raise error earlier

clemkoa committed:
[MRG+1] Fixing Math domain error on NMF due to numpy.dot (#8765)

* Fixing overflow error on NMF due to numpy.dot

petebachant committed:
[MRG+1] De-italicize non-variables in formulas (#8901)

* De-italicize non-variables in formulas

* Fix some TeX syntax errors

qinhanmin2014 committed:
Fix n_splits in KFold instantiation in model_selection tests (#8910)

jorisvandenbossche committed:
DOC: styling of deprecated boxes (#8914)

qinhanmin2014 committed:
Update model_evaluation.rst (#8915)

fujiyuu75 committed:
Fix typo in PCA test (#8917)

glemaitre committed:
[MRG] DOC Homogeneous deprecation in docstring (#8907)

Update developers doc to use "deprecated" sphinx directive.

massich committed:
[MRG] DOC add documentation on adding non-regression tests (#8850)

lesteve committed:
DOC add plot_stock_market.py to expected failing examples (#8921)

as a stop gap solution until we figure out a longer-term work-around.

Also remove nibabel from conf.py.

Klysium committed:
Fixed markdown in doc/README.md (#8925)

jnothman committed:
[MRG+1] Update version dependencies to numpy 1.8.2, scipy 1.13.3 (#8855)

naoyak committed:
Update numpy refs to 1.8.2 (#8926)

Commit:
documentation fix (#8928)

Commit:
[MRG] documentation fix in neighbors module (#8929)

Attractadore committed:
[MRG+1] Added n_components parameter to LatentDirichletAllocation to replace … (#8922)

[MRG+2] Added n_components parameter to LatentDirichletAllocation to replace …

jmschrei committed:
Revert "[MRG+1] Added n_components parameter to LatentDirichletAllocation to replace … (#8922)" (#8937)

This reverts commit 890e65289877db32c0c440cadaa2b2ef2ef7a384.

Attractadore committed:
[MRG+1] Rename n_topics to n_components in LDA (#8938)

jnothman committed:
DOC update whats_new with LDA n_topics deprecation

joernhees committed:
ENH mlp convergence warning text (#8947)

jnothman committed:
BUILD Give clear error message when there are merge conflicts (#8945)

Terryhung committed:
[MRG+1] Add set cardinality formulation for mutual_info_score (#8821)

* Remove unnecessary braces.

* Replace R, C with |U|, |V|.

* Remove the probability formulation from docstring.

qinhanmin2014 committed:
[MRG] DOC add function load_wine to the document (#8965)

* add wine dataset to the document

* Update classes.rst

* Update index.rst

chinmayapancholi13 committed:
DOC fixed typo in online_lda.py (#8969)

deto committed:
[MRG] Fix gradient descent in TSNE (#8768)

Skip flaky test_n_iter_without_progress on 32bit

qinhanmin2014 committed:
[MRG+1] DOC use a table in load_linnerud docstring (#8970)

jnothman committed:
[MRG+2] ENH Loop over candidates as outer loop in search (#8322)

This encourages concurrent fits to be over *different datasets* so that
fits over the same data subset are more likely to run in serial and
hence generate cache hits where memoisation is used.

Commit:
Updated error message for max_leaf_nodes (#8975)

Solves #8964

naoyak committed:
[MRG+1] MAINT drop SciPy < 0.13 (#8854)

Remove sklearn.utils.fixes functions that are not needed for scipy >= 0.13 and keep deprecated wrappers in other modules.

albertcthomas committed:
DOC recommend editable install using pip in contributing (#8974)

taehoonlee committed:
Fix typos (#8980)

reiinakano committed:
[MRG+1] Documentation: Add xcessiv to related projects (#8985)

* Documentation: Add xcessiv to related projects

For your consideration, this adds Xcessiv to the list of related projects under experimentation frameworks. Xcessiv lets you fully define your data, cross-validation technique, estimators, and metrics and is based heavily on the scikit-learn API.

* Modify second sentence

qinhanmin2014 committed:
[MRG+1] Change the restriction of n_repeats in RepeatedKFold and RepeatedStratifiedKFold (#8909)

* change the restriction of n_repeats

* change test case

shivamgargsya committed:
DOC add reference in probabilities to SVC doc (#8990)

dohmatob committed:
BF: fixing error on plot_image_denoising.py with Python 2.7 (#9003)

souravsingh committed:
[#8972] Add links to FAQ (#9002)

* Add links to FAQ

* Update faq.rst

* Fix typo

mblondel committed:
Division.

amueller committed:
[MRG+1] Uncontroversial fixes from estimator tags branch (#8086)

* some bug fixes.

* minor fixes to whatsnew

* typo in whatsnew

* add test for n_components = 1 transform in dict learning

* feature extraction doc fix

* fix broken test

* revert aggressive input validation changes

* in SelectFromModel, don't store threshold_ in transform. If we called "fit", use estimates from last "fit".

* move score from EllipticEnvelope to OutlierDetectionMixin

* revert changes to Tfidf documentation

* remove dummy input validation from whatsnew

* fix text feature tests

* rewrite from_model threshold again...

* remove stray condition

* fix self.estimator -> estimator, slightly more interesting test

* typo in comment

* Fix issues in SparseEncoder, add tests.
more explicit explanation of SparseEncoder change, add issue numbers to whatsnew

* minor fixes in whats_new.rst

* slightly more consistency with tuples for shapes

* not longer typo

Titan-C committed:
Update to Sphinx-Gallery v0.1.11 (#8222)

taehoonlee committed:
Fix typos in test_covariance.py and extmath.py (#8997)

jnothman committed:
[MRG+1] LDA verbose should print iteration number in all cases (#8944)

* Printing iter and max_iter in online_lda if verbose

* Fixed typo to print iter and max_iter

* TST test verbose feature of LatentDirichletAllocation

* Py2 compatibility

GaelVaroquaux committed:
DOC: Simpler precision-recall example, remove 11pt

Remove the eleven average precision score

Add better tests.

remram44 committed:
Fix link to cross_validation (#9025)

Previously read "search [...] for the best Cross-validation: evaluating estimator performance score", because the title of 'cross_validation' changed.

superbobry committed:
ENH use expit in ``_BinaryGaussianProcessClassifierLaplace`` (#9011)

SciPy already provides a numerically stable ``special.expit``
function, which does not overflow on the example mentioned in #8641.

paulochf committed:
[MRG] Fix LassoCV cross validation split() call (#8973)

* Fixing cross validation split call in LassoCV

* Non-regression test for LassoCV cv.split check

* Fix typo KFold->StratifiedKFold

superbobry committed:
[MRG] Updated plot_stock_market.py to use Google Finance (#9010)

* DOC updated plot_stock_market.py to use Google Finance

The implementations is intentionally very basic not to distract the users
from the example. Specifically unlike ``quotes_historical_yahoo_ochl`` it
does not cache downloaded data.

I also had to remove some symbols because the have no data on Google for
the specified date interval. These are WBA, LMT, KFT and MTU.

Closes #8899

* DOC removed plot_stock_market.py from expected failing examples

* Addressed review comments

* Addressed another pass of review comments

agramfort committed:
plt import standard + pep8

amueller committed:
fix multioutput partial_fit delegation (#9013)

massich committed:
[MRG+2] LogisticRegression convert to float64 (newton-cg) (#8835)

* Add a test to ensure not changing the input's data type

Test that np.float32 input data is not cast to np.float64 when using LR + newton-cg

* [WIP] Force X to remain float32. (self.coef_ remains float64 even if X is not)

* [WIP] ensure self.coef_ same type as X

* keep the np.float32 when multi_class='multinomial'

* Avoid hardcoded type for multinomial

* pass flake8

* Ensure that the results in 32bits are the same as in 64

* Address Gael's comments for multi_class=='ovr'

* Add multi_class=='multinominal' to test

* Add support for multi_class=='multinominal'

* prefer float64 to float32

* Force X and y to have the same type

* Revert "Add support for multi_class=='multinominal'"

This reverts commit 4ac33e8c02afb87160a6360f6624d52398edfe5c.

* remvert more stuff

* clean up some commmented code

* allow class_weight to take advantage of float32

* Add a test where X.dtype is different of y.dtype

* Address @raghavrv comments

* address the rest of @raghavrv's comments

* Revert class_weight

* Avoid copying if dtype matches

* Address alex comment to the cast from inside _multinomial_loss_grad

* address alex comment

* add sparsity test

* Addressed Tom comment of checking that we keep the 64 aswell

agramfort committed:
update what's new

superbobry committed:
FIX memory leak in liblinear (#9024)

The leak resulted from two issues:
- not freeing the problem struct
- not freeing the number of iterations

The former was present in the initial version of ``liblinear_helper.c``
while latter appeared after c8c72fd96e13f9a7fd80362311248321949b8de5
which introduced ``n_iter``.

Closes #8499

jnothman committed:
DOC update what's new

lesteve committed:
TRAVIS only install apt packages when needed (#9016)

Also revamped .travis.yml to use the matrix/include for all build entries

abhyudaypratap committed:
LogisticregressionCV docs(#8424): documented C_ shape (#8434)

GaelVaroquaux committed:
DOC: fix whats_new link

Fixes #8885

lesteve committed:
TRAVIS temporarily remove pytest and scipy-dev build

as an attempt to reduce the Travis queue

Sentient07 committed:
[MRG+1] Removed warning from BaseSearchCV (#9005)

Commit:
[MRG + 1] 18 more examples with matplotlib 2.0 updates (#8983)

* updated plot_label_propagation_versus_svm_iris.py plot

* updated svm/plot_weighted_samples.py plot

* made semi_supervised/plot_label_propagation_versus_svm_iris.py pep8 compliant

* modified tree/plot_tree_regression.py [size and edgecolor]

* updated tree/plot_tree_regression_multioutput.py [size+color]

* fixed examples/semi_supervised/plot_label_propagation_versus_svm_iris.py for backward compatibility

* neural_networks/plot_mlp_alpha.py - matplotlib2 update

* examples/neural_networks/plot_mlp_alpha.py - pep8 fix

* examples/neighbors/plot_nearest_centroid.py - matplotlib2.0 + pep8 fix

* neighbors/plot_classification.py - matplotlib2.0 + pep8 fix

* examples/neighbors/plot_lof.py - matplotlib2.0 update

* examples/model_selection/plot_underfitting_overfitting.py - matplotlib2.0 + pep8

* examples/mixture/plot_concentration_prior.py - matplotlib2.0 + pep8

* examples/linear_model/plot_logistic_multinomial.py - matplotlib2.0 update

* linear_model/plot_sgd_iris.py - matplotlib2.0 + pep8 fix

* examples/linear_model/plot_sgd_weighted_samples.py - matplotlib2.0 + pep8

* examples/linear_model/plot_sgd_separating_hyperplane.py - matplotlib2.0 update

* examples/feature_selection/plot_permutation_test_for_classification.py - matplotlib + pe8

* examples/linear_model/plot_bayesian_ridge.py - matplotlib2.0 update

* examples/feature_selection/plot_feature_selection.py - matplotlib2.0 update

* examples/feature_selection/plot_f_test_vs_mi.py - matplotlib2.0 + pep8

* examples/feature_selection/plot_f_test_vs_mi.py - matplotlib2.0+ pep8 fix

* examples/model_selection/plot_underfitting_overfitting.py - error fixed

* blue -> black edgecolor fix for 2 examples

naoyak committed:
[MRG+1] Drop NumPy < 1.8 (#8874)

dalmia committed:
[MRG+1] Better error message for GPR (#8386)

* FIX: better error message for GPR

* TST: added tests for error message

* FIX: correct error message and simplified tests

* DOC: updated docstring for GPR

* FIX: pass float as value for alpha

* Raise original error after modification to preserve traceback

Amend test accordingly

bellet committed:
MRG: add warning/doc for positive parameter in multi output (#9035)

* ENH: add warning/doc for positive parameter in multi output

added warning and updated doc to cover multi output case where positive parameter is ignored

* change warning into ValueError

* create single dataset for test_enet_path_positive

amanp10 committed:
[MRG+1] Issue#5803 : Regression Test added (#8112)

* Issue#5803 Regression Test added

* Float predictions values used for testing

* Custom Classifier used for test

agramfort committed:
better comment in test

dalmia committed:
[MRG+1] ENH: added max_train_size to TimeSeriesSplit (#8282)

* ENH: added max_train_size to TimeSeriesSplit

* FIX: update doctest

* FIX: correct error in the previous update

* FIX: added doctest fix for cross_validation.rst

* FIX: remove errors

* TST: tests updated and default value changed to None

* TST: improve split tests

* FIX: reduce code length

agramfort committed:
update + cleanup what's new

mlewis1729 committed:
[MRG+1] fixed OOB_Score bug for bagging classifiers. (#8936)

* fixed OOB_Score bug for bagging slassifiers.
See: https://github.com/scikit-learn/scikit-learn/issues/8933

* Added white space

* more white space fixing

* Adding test for oob_score validity

* removing pandas, replacing with numpy matrices

* fixing white space

* more white space fixing

* white space ...

* fixed labels to allow for strings

* white space

* simplifying test

* white space

* reformatting test

* white space

* pressed enter at end of file

* removing line at end of file

agramfort committed:
pep8 + update what's new

Commit:
[MRG+1] add shuffle paramater to train_test_split (#8845)

* add shuffle paramater to train_test_split

* fix syntax error

* fix variable name

* fix formatting in doctest output

* fix doctest output

* refactor shuffle paramater into ShuffleSplit and StratifiedShuffleSplit

* include shuffle option in tests

* rollback refactor

* revert to simpler version of unshuffled split

* fix flake8 errors

* revert changes to ShuffleSplit

* revert BaseShuffleSplit

* more reversions

* fix indentation

* remove shuffle parameter from CVclass

* add text to NotImplementedError

* change indexing to use numpy.arange rather than range

* specify precondition for stratify to be None in docstring

agramfort committed:
use ValueError

agramfort committed:
update what's new

ThatGeoGuy committed:
[MRG+1] Fixes issue #3367 -> MCD fails on data with singular covariance matrix (#8328)

* Adds failing test for issue 3367

* Adds extra comments to describe what I'm testing

Basically in this instance I discovered this bug independently from
\#3367 because I was trying to use principle components to estimate
plane normals / geometry of points in 3D space. When you have a set of
points that specify a perfect plane though (or in the case of wanting to
use MCD, you have a subset of your points that specify a perfect plane),
then the code fails because the covariance matrix is singular. However,
if your covariance matrix is singular, you've already found the set of
points with the lowest determinant. As per Rousseeuw & Van Driessen
1999, at this point you can stop searching.

The code did stop searching, however, it raised a ValueError on singular
matrices for no reason. So the correct fix should be to remove that.

* Fixes issue 3367

This should work with the test case provided.

* Adds missing argument to test

* Style corrections to pass flake runner

Implements the style corrections as mentioned in pull request #8328

* Adds entry for PR #8328 to what's new

* Adds review changes from @jnothman

agramfort committed:
pep8 + oups...

davidgasquez committed:
[MRG] Add DCG and NDCG (#7739)

* Add DCG and NDCG ranking functions

rth committed:
[MRG] FIX Hash collisions in the FeatureHasher (#7565)

* HashingVectorizer: optionaly disable alternate signs

jnothman committed:
[MRG+1] Option to suppress validation for finiteness (#7548)

* ENH add suppress validation option

* TST skip problematic doctest

* Rename SUPPRESS_VALIDATION to PRESUME_FINITE

* Change PRESUME_ to ASSUME_ for convention's sake

* DOC add note regarding assert_all_finite

* ENH add set_config context manager for ASSUME_FINITE

* Make ASSUME_FINITE private and provide get_config

* Fix ImportError due to incomplete change in last commit

* TST/DOC tests and more cautious documentation for set_config

* DOC what's new entry for validation suppression

* context manager is now config_context; set_config affects global config

* Rename missed set_config to config_context

* Fix mis-named test

* Mention set_config in narrative docs

* More explicit about limmited restoration of context

* Handle case where error raised in config_context

* Reset all settings after exiting context manager

chribsen committed:
Change short description of predict_proba in sklearn.mixture (#8690) (#8719)

amueller committed:
fix biclustering API (#9053)

don't mention returning self in whatsnew as that's already fixed in master

agramfort committed:
[MRG + 1] Ovr cleanup (#9051)

* fix test for new default of SVC decision function

* == not is

herilalaina committed:
[MRG+1] Make classification (dimensions > 30) (#9045)

* Change _generate_hypercube into rng.randint

* Improve unit test

* Test if each row is unique

albertcthomas committed:
[MRG+1] DOC clarify OneClassSVM decision_function docstring (#9048)

* clarify ocsvm decision_function docstring

* take into account agramfort comment

glemaitre committed:
DOC add docstring to FunctionTransformer (#9058)

herilalaina committed:
Update what's new for fixing make_classification (#9060)

jnothman committed:
DOC List more utils in API ref (#8827)

jnothman committed:
FIX Avoid unintentionally converting numeric data to dtype=object (#9049)

lesteve committed:
Fix estimators_checks.test_non_meta_estimators

Partially reverts 9a5e160

vene committed:
DOC fix markdown-style link and some wording (#9065)

agramfort committed:
[MRG+1] y not needed in transform in Birch (#9066)

* y not needed in transform

* y not needed in transform

* more

lesteve committed:
FIX examples with numpy 1.13

qinhanmin2014 committed:
[MRG] DOC Correct function name KDtree -> KDTree (#9072)

* function name KDtree -> KDTree

* correct similar mistakes

qinhanmin2014 committed:
fix doc (#9074)

lesteve committed:
[MRG] Use ccache on Travis (#9006)

mrbeann committed:
Add ref (#9064)

* Add ref

* Fix pep8

jayzed82 committed:
Fix PLS scaling bug (#7819)

amueller committed:
[MRG+1] don't set fit_params in __init__ in BaseSearchCV (#9038)

* don't set fit_params in __init__ in BaseSearchCV

* fix doctest

lesteve committed:
COSMIT add missing spaces in deprecated messages

vene committed:
[MRG+2] MAINT: no longer backport graph_laplacian, use scipy one (#9077)

ldirer committed:
Fix decorator called without kwarg that would prevent test from running. (#9083)

amueller committed:
[MRG+1] Instance level common tests (#9019)

* start work on separating instance-level tests

* minor refactoring / fixes to work without tags

* add clone into check_supervised_y_2d estimator check (which made other checks fail)

* remove duplicate check_estimator_unfitted assert

* add issue reference to whatsnew entry

* added some clones, minor fixes from vene's review

* rename estimator arg to estimator_org to make a visible distinction before and after cloning.

* more renaming for more explicit clones

* org -> orig

* allclose, fix orig stuff

* don't use set_testing_parameters in the checks!

* minor fixes for allclose

* fix some test, add more tests on classes

* added the test using pickles.

* move assert_almost_equal_dense_sparse to utils.testing, rename to assert_allclose_sparse_dense, test it

* make assert_allclose_dense_sparse more stringent

* more allclose fixes

* run test_check_estimator on all estimators

* rename set_testing_parameters to set_checking_parameters so nose doesn't think it's a tests (and I don't want to import stuff from nose as we want to remove it)

* fix in set_checking_parameters so that common tests pass

* more fixes to assert_allclose_dense_sparse

* rename alg to clusterer, don't scream even though I really want to

* ok this is not a pretty strict test that runs check_estimator with and without fitting on an instance. I also check if ``fit`` is called on the instance that is passed.

* simplify test as they didn't help at all

* it works!!! omfg

* run check_estimator clone test only on one of the configs, don't run locally by default

* Add `slow_test` decorator and documentation

* run test_check_estimator only on some estimators

* fix diags in test for older scipy

* fix pep8 and shorten

* use joblib.hash for inequality check because the pickle state machine is weird

lesteve committed:
COSMIT naming improvement

Attractadore committed:
Fixed sklearn-related invalid escape sequence DesprecationWarnings (#8951)

ngoix committed:
[MRG+2] clean outlier_detection.py (#9018)

Remove OutlierDetectionMixin, which was only used by by EllipticEnvelope

glemaitre committed:
[MRG+1] QuantileTransformer (#8363)

* resurrect quantile scaler

* move the code in the pre-processing module

* first draft

* Add tests.

* Fix bug in QuantileNormalizer.

* Add quantile_normalizer.

* Implement pickling

* create a specific function for dense transform

* Create a fit function for the dense case

* Create a toy examples

* First draft with sparse matrices

* remove useless functions and non-negative sparse compatibility

* fix slice call

* Fix tests of QuantileNormalizer.

* Fix estimator compatibility

* List of functions became tuple of functions
* Check X consistency at transform and inverse transform time

* fix doc

* Add negative ValueError tests for QuantileNormalizer.

* Fix cosmetics

* Fix compatibility numpy <= 1.8

* Add n_features tests and correct ValueError.

* PEP8

* fix fill_value for early scipy compatibility

* simplify sampling

* Fix tests.

* removing last pring

* Change choice for permutation

* cosmetics

* fix remove remaining choice

* DOC

* Fix inconsistencies

* pep8

* Add checker for init parameters.

* hack bounds and make a test

* FIX/TST bounds are provided by the fitting and not X at transform

* PEP8

* FIX/TST axis should be <= 1

* PEP8

* ENH Add parameter ignore_implicit_zeros

* ENH match output distribution

* ENH clip the data to avoid infinity due to output PDF

* FIX ENH restraint to uniform and norm

* [MRG] ENH Add example comparing the distribution of all scaling preprocessor (#2)

* ENH Add example comparing the distribution of all scaling preprocessor

* Remove Jupyter notebook convert

* FIX/ENH Select feat before not after; Plot interquantile data range for all

* Add heatmap legend

* Remove comment maybe?

* Move doc from robust_scaling to plot_all_scaling; Need to update doc

* Update the doc

* Better aesthetics; Better spacing and plot colormap only at end

* Shameless author re-ordering ;P

* Use env python for she-bang

* TST Validity of output_pdf

* EXA Use OrderedDict; Make it easier to add more transformations

* FIX PEP8 and replace scipy.stats by str in example

* FIX remove useless import

* COSMET change variable names

* FIX change output_pdf occurence to output_distribution

* FIX partial fixies from comments

* COMIT change class name and code structure

* COSMIT change direction to inverse

* FIX factorize transform in _transform_col

* PEP8

* FIX change the magic 10

* FIX add interp1d to fixes

* FIX/TST allow negative entries when ignore_implicit_zeros is True

* FIX use np.interp instead of sp.interpolate.interp1d

* FIX/TST fix tests

* DOC start checking doc

* TST add test to check the behaviour of interp numpy

* TST/EHN Add the possibility to add noise to compute quantile

* FIX factorize quantile computation

* FIX fixes issues

* PEP8

* FIX/DOC correct doc

* TST/DOC improve doc and add random state

* EXA add examples to illustrate the use of smoothing_noise

* FIX/DOC fix some grammar

* DOC fix example

* DOC/EXA make plot titles more succint

* EXA improve explanation

* EXA improve the docstring

* DOC add a bit more documentation

* FIX advance review

* TST add subsampling test

* DOC/TST better example for the docstring

* DOC add ellipsis to docstring

* FIX address olivier comments

* FIX remove random_state in sparse.rand

* FIX spelling doc

* FIX cite example in user guide and docstring

* FIX olivier comments

* EHN improve the example comparing all the pre-processing methods

* FIX/DOC remove title

* FIX change the scaling of the figure

* FIX plotting layout

* FIX ratio w/h

* Reorder and reword the plot_all_scaling example

* Fix aspect ratio and better explanations in the plot_all_scaling.py example

* Fix broken link and remove useless sentence

* FIX fix couples of spelling

* FIX comments joel

* FIX/DOC address documentation comments

* FIX address comments joel

* FIX inline sparse and dense transform

* PEP8

* TST/DOC temporary skipping test

* FIX raise an error if n_quantiles > subsample

* FIX wording in smoothing_noise example

* EXA Denis comments

* FIX rephrasing

* FIX make smoothing_noise to be a boolearn and change doc

* FIX address comments

* FIX verbose the doc slightly more

* PEP8/DOC

* ENH: 2-ways interpolation to avoid smoothing_noise

Simplifies also the code, examples, and documentation

TomDLT committed:
FIX clean whats_new (#9090)

djsutherland committed:
clarify max_iter meaning for MLPClassifier/MLPRegressor (#9089)

taehoonlee committed:
Fix typos (#9095)

naoyak committed:
Add logsumexp and comb to utils.fixes (#9046)

glemaitre committed:
[MRG+1] DOC add hyperlink to example (#9097)

* DOC add hyperlink to example

* Remove useless change

* DOC fix hyperlink

* DOC fix links

ldirer committed:
[MRG] Deprecate lsh forest (#9078)

* Add deprecation message and test.

* Adding performance warning and ignore_warnings in test

* Add deprecation to whatsnew and remove LSHForest references from docs.

Removing benchmark for lsh

agramfort committed:
[MRG+1] remove n_nonzero_coefs from attr of LassoLarsCV + clean up call hierarchy (#9004)

* FIX : remove n_nonzero_coefs from attr of LassoLarsCV + clean up call to Lars._fit

* cleanup

* fix deprecation warning + clarify warning

* add test

* pep8

* adddress comments

agramfort committed:
add doc + refs + what's new entry (#9052)

amueller committed:
FIX OvR/OvO classifier decision_function shape fixes (#9100)

* fix OVR classifier edgecase bugs

* add regression tests for OVO and OVR decision function shapes

glemaitre committed:
[MRG + 1] FIX/TST revert #5802 and raise error for faulty classifier (#9063)

* FIX/TST revert #5802 and raise error for faulty classifier

* FIX check_estimator take care of the rest

agramfort committed:
[FIX] BIC/AIC for Lasso (#9022)

* correcting information criterion calculation in least_angle.py

The information criterion calculation is not compatible with the original paper 
Zou, Hui, Trevor Hastie, and Robert Tibshirani. "On the “degrees of freedom” of the lasso." The Annals of Statistics 35.5 (2007): 2173-2192.
APA

* FIX : fix AIC/BIC computation in LassoLarsIC

* update what's new

* fix test

* fix test

* address comments

* DOC comments and docstring on criterion computation

vene committed:
DOCFIX typo & pep8 & shame

lesteve committed:
CIRCLE latexmk is needed to build the pdf doc

with sphinx 1.6

lesteve committed:
Disable pager in git commands

Long commit messages can trigger a pager which is not what you want
when running flake8_diff.sh in a terminal.

jorisvandenbossche committed:
DOC: fix links to examples (#9102)

lesteve committed:
CIRCLE Revert to sphinx 1.5

to fix pdf doc generation

lesteve committed:
TRAVIS put back all the builds

now that the sprint is over.

jnothman committed:
What's new addition and fixes

ClimbsRocks committed:
DOC adds auto_ml to related projects (#9042)

albertcthomas committed:
remove identical assert in test_iforest_sparse (#9112)

TomDLT committed:
FIX improve 'precompute' handling in Lars (#5359)

Attractadore committed:
Removed force_all_finite array checks in DummyClassifier and DummyRegressor, added 2 tests (#8931)

vene committed:
[MRG+1] fix StratifiedShuffleSplit with 2d y (#9044)

* regression test and fix for 2d stratified shuffle split

* strengthen non-overlap sss tests

* clarify test and comment

* remove iter from tests, use str instead of hash

jnothman commented:
This only needs a little fixing up to be merged and included in the upcoming release. @Sundrique, are you onto it? Please also add an entry to `doc/whats_new.rst`

Sundrique commented:
@jnothman Yes, I am. Will add into `model_selection/_validation.py`, resolve the conflict and add an entry to `doc/whats_new.rst`. Please, let me know if anything else is needed.

jnothman commented:
Hopefully after those changes, @ogrisel can give this a final review and
merge.

On 14 June 2017 at 11:09, Aleksandr Sandrovskii <notifications@github.com>
wrote:

> @jnothman <https://github.com/jnothman> Yes, I am. Will add into
> model_selection/_validation.py, resolve the conflict and add an entry to
> doc/whats_new.rst. Please, let me know if anything else is needed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7365#issuecomment-308290904>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz69DbZejMdY3JbhEs1a6Vr9BaQSntks5sDzK1gaJpZM4J4CRC>
> .
>

Sundrique committed:
Restore mock estimator after rebase

Sundrique committed:
Apply changes to the model_selection module

Sundrique committed:
Update whats_new

Sundrique commented:
@jnothman Sorry, screwed up the branch. Created a new one #9119.
